erdos/src/dataflow/stream/loop_stream.rs

```diff
@@ -26,34 +29,27 @@ where
     for<'a> D: Data + Deserialize<'a>,
 {
     id: StreamId,
+    name: String,
```

Comment:
Loop streams also shouldn't have a separate name. They should inherit the name from whichever stream they're connected to when the loop is closed. Otherwise the following presents a consistency issue:
```rust
let loop_stream = graph.add_loop_stream("s1");`
let stream = loop_stream.map(|x| 2 * x);
stream.set_name("s2");
loop_stream.connect(&stream);
```

Reply:
I agree, we should probably not name the `LoopStream` since they are just virtual streams that will be removed after the graph's `compile` method is called.

Reply:
Counterpoint: it is useful to name loops for debugging (e.g. when a loop is unconnected). Also, if we apply LINQ operations to a loop stream, we might have trouble naming the LINQ operators.